packit/api.py

```diff
@@ -383,22 +399,27 @@ def sync_from_downstream(
             self.up.checkout_branch(local_pr_branch)
 
         files = (
-            SyncFilesConfig([self.package_config.get_specfile_sync_files_item()])
+            [self.package_config.get_specfile_sync_files_item()]
             if sync_only_specfile
             else self.package_config.synced_files
         )
 
-        raw_sync_files = files.get_raw_files_to_sync(
-            dest_dir=self.dg.local_project.working_dir,
-            src_dir=self.up.local_project.working_dir,
-        )
-
-        reverse_raw_sync_files = [
-            raw_file.reversed()
-            for raw_file in raw_sync_files
-            if Path(raw_file.dest).name not in exclude_files
-        ]
-        sync_files(reverse_raw_sync_files, fail_on_missing=False)
+        # Drop files to be excluded from the sync.
+        for ef in exclude_files:
+            files = [
+                f.drop_src(ef, criteria=lambda x, y: x.name == y)
+                for f in files
+                if f is not None
+            ]
```

Comment:
I am thinking if it makes sense to have some abstraction around those functional code snippets using the list of files.

Reply:
What exactly you mean by this? :thinking:

Reply:
What I don't like?
* There are code parts, that run some functional-style code on the list of excluded files. And those code parts are really dense and hard to read.
* A lot goes on in one place. (But I am glad that there are comments with the real purpose.)

What am I suggesting?
* Class that will "hide" this? Some well-named helper method/function?

Reply:
You mean to wrap the list of synced_files in a class?

I thought about that. The reason I gave up on that is that it would create an indirection (one would need to navigate and read the method to see what's doing), which would be okay, if it would help with de-duplicating code, but with the current usage that wouldn't be the case. Adding a class would increase complexity while resulting in very little benefit.

Reply:
> You mean to wrap the list of synced_files in a class?

Something like that, or some function(s)/method(s).

With a good name, I don't see a problem with an indirection. That's just a decomposition. But that's nothing crucial. If you have thought about that and think it's not needed, let's leave it as is.

Reply:
:+1: Thanks!